src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -1157,24 +1180,30 @@ def set_tags(self, tags):
         lv_object = get_lv(lv_name=self.lv_name, lv_path=self.lv_path)
         self.tags = lv_object.tags
 
+
+    def clear_tag(self, key):
+        if self.tags.get(key):
+            current_value = self.tags[key]
+            tag = "%s=%s" % (key, current_value)
+            process.call(['lvchange', '--deltag', key, self.lv_api['lv_path']])
+
+
     def set_tag(self, key, value):
         """
         Set the key/value pair as an LVM tag. Does not "refresh" the values of
         the current object for its tags. Meant to be a "fire and forget" type
         of modification.
         """
         # remove it first if it exists
-        if self.tags.get(key):
-            current_value = self.tags[key]
-            tag = "%s=%s" % (key, current_value)
-            process.call(['lvchange', '--deltag', tag, self.lv_api['lv_path']])
+        self.clear_tag(key)
 
         process.call(
             [
                 'lvchange',
                 '--addtag', '%s=%s' % (key, value), self.lv_path
             ]
         )
+        self.tags[key] = value
```

Comment:
@alfredodeza this might have been a bug. `self.tags` wasn't updated as tags were set. Can you confirm?

Reply:
I guess it depends on how you see this `clear_tag` feature being used. In the `set_tag` case, we don't necessarily need to check again tags, we are mainly adding tags and then we are done with it. 

In the few situations were we do refresh the tags, it is because we checked against those later and we would be surprised if the object doesn't reflect what exists in reality.

Reply:
I understand. If there's no objection, I'll just keep the update in `clear_tag` so that the `tags` object reflects the state of the volume.

Do you want me to revert this update here in `set_tag`?

Reply:
I think this looks fine, and I am OK with it if it doesn't break anything in the functional test suite :)

Reply:
@alfredodeza great, thanks :) I've updated this entire PR, I think it's ready for review.

Reply:
actually, not yet. some tests are failing for python2, looking.

Reply:
Really sorry for all the noise. Now all tests pass with python2 and python3. The issue was that with python2, the order of parsing the dictionary wasn't guaranteed, the so the order of the commands in `expected` was different, although the same commands are executed. Anyway, this PR is ready for reviewing.